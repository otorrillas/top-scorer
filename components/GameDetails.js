import React from 'react';
import PropTypes from 'prop-types';

const GameDetails = ({ time, video, player }) => {
  const minutes = Math.floor(time / 60);
  const seconds = time % 60;

  return (
    <>
      <p>
        <b>Time:</b> {minutes} minutes and {seconds} seconds
      </p>
      <p>
        <b>Player:</b> {player}
      </p>
      <button>
        <a href={video} target="_blank" rel="noopener noreferrer">
          Watch run video
        </a>
      </button>
    </>
  );
};

GameDetails.propTypes = {
  time: PropTypes.number.isRequired,
  video: PropTypes.string.isRequired,
  player: PropTypes.string.isRequired
};

export default GameDetails;
