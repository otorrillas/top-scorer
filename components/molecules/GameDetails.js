import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import _find from 'lodash/head';
import _head from 'lodash/head';

import { getGames } from 'Store/selectors';
import { GameShape } from 'Utils/propTypes';

import TotalTime from '../atoms/TotalTime';

export const GameDetailsView = ({ games, id }) => {
  const game = games[id];
  const { abbreviation: name, assets, runs } = game;

  const {
    videos,
    players,
    times: { primary_t }
  } = _head(runs);

  const playerName = _head(players).id;
  const videoSrc = _head(videos.links).uri;
  const logoSrc = assets['cover-medium'].uri;

  return (
    <>
      <h1>{name}</h1>
      <img src={logoSrc} alt={`${name} logo`} />

      <p>
        <b>Time:</b> <TotalTime time={primary_t} />
      </p>
      <p>
        <b>Player:</b> {playerName}
      </p>

      <button>
        <a href={videoSrc} target="_blank" rel="noopener noreferrer">
          Watch run video
        </a>
      </button>
    </>
  );
};

GameDetailsView.propTypes = {
  games: PropTypes.objectOf(GameShape).isRequired,
  id: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  games: getGames(state)
});

export default connect(
  mapStateToProps,
  null
)(GameDetailsView);
