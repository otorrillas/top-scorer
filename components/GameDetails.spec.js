import React from 'react';
import renderer from 'react-test-renderer';

import GameDetails from './GameDetails';

describe('<GameDetails />', () => {
  const renderComponent = props => renderer.create(<GameDetails {...props} />);

  describe('@render', () => {
    it('default', () => {
      const tree = renderComponent({
        time: 1234,
        video: 'https://someVideoUrl',
        player: 'SomePlayer'
      });
      expect(tree).toMatchSnapshot();
    });
  });
});
