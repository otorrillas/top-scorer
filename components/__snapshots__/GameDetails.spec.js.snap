// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameDetails /> @render default 1`] = `
Array [
  <p>
    <b>
      Time:
    </b>
     
    20
     minutes and 
    34
     seconds
  </p>,
  <p>
    <b>
      Player:
    </b>
     
    SomePlayer
  </p>,
  <button>
    <a
      href="https://someVideoUrl"
      rel="noopener noreferrer"
      target="_blank"
    >
      Watch run video
    </a>
  </button>,
]
`;
